lib/canvas/live_events.rb

```diff
@@ -101,6 +101,23 @@ def self.get_submission_data(submission)
     }
   end
 
+  def self.get_attachment_data(attachment)
+    {
+      attachment_id: attachment.global_id,
+      user_id: attachment.global_user_id,
+      display_name: LiveEvents.truncate(attachment.display_name),
+      filename: LiveEvents.truncate(attachment.filename),
+      context_type: attachment.context_type,
+      context_id: attachment.context_id,
+      content_type: attachment.content_type,
+      folder_id: attachment.folder_id,
```

Comment:
Cool, that should simplify a few things (and minimise the number of API requests we'll have to make)